Python/bltinmodule.c

```diff
@@ -2628,24 +2643,67 @@ zip_next(zipobject *lz)
             item = (*Py_TYPE(it)->tp_iternext)(it);
             if (item == NULL) {
                 Py_DECREF(result);
+                if (lz->tuplesize < 0) {
+                    goto check;
+                }
                 return NULL;
             }
             PyTuple_SET_ITEM(result, i, item);
         }
     }
     return result;
+check:
+    if (PyErr_Occurred() && !PyErr_ExceptionMatches(PyExc_StopIteration)) {
+        return NULL;
+    }
+    if (i) {
+        PyErr_Clear();
+        return PyErr_Format(PyExc_ValueError, "%s() argument %d is too short",
```

Comment:
It bothers me that the implementation gives different errors depending on whether it's the first iterator that's longer or any of the later ones. It seems to assume that the first iterator is "right" and the others are either too short or too long. But isn't it more the case that all we know is that they aren't all of the same length -- we shouldn't judge which one is too long or too short. Perhaps the errors could be changed to something like "iterator N is shorter/longer than the preceding iterators"?

Reply:
Yeah, perhaps these are a bit too terse.  I'll think on some other options (I also agree that unifying the messages is probably a good idea).